bothan-lib/src/metrics/rest.rs

```diff
@@ -0,0 +1,86 @@
+use opentelemetry::metrics::{Counter, Histogram};
+use opentelemetry::{KeyValue, global};
+
+pub enum RequestStatus {
+    Success,
+    Failed,
+    Timeout,
+}
+
+impl RequestStatus {
+    fn as_str_name(&self) -> &'static str {
+        match self {
+            RequestStatus::Success => "success",
+            RequestStatus::Failed => "failed",
+            RequestStatus::Timeout => "timeout",
+        }
+    }
+}
```

Comment:
as_str() should be sufficient. If we don't need the static lifetime implement Display instead